TODO/v8-behind-the-scenes-november-edition.md

```diff
@@ -105,11 +119,11 @@ function A() { ... }
 function isA(o) { return o instanceof A; }
 ```
 
-would be slowed down a lot if you implement ES2015 naively, because in addition to the actual prototype chain walk that you need to perform for `instanceof`, you know also need to lookup the `@@hasInstance` property on `A`’s prototype first and call that method. To mitigate that problem we decided to go with a mechanism called a *protector cell* in the beginning, which allows certain parts of V8 to assume that a certain event didn’t happen so far, so we don’t need to perform certain checks. In this case the protector cell would guard against the addition of a property named `Symbol.hasInstance` anywhere in V8. Assuming that no one installed any other `@@hasInstance` function anywhere, we could just continue to always implement `instanceof` via OrdinaryHasInstance as long as the protector is intact.
+如果采用 ES2015 的方式，函数性能将大打折扣，因为运行 `instanceof` 不光要追踪原型链产生额外开销，还需要检查  `A `  的原型链上是否有  `@@hasInstance`  属性以便调用。为了降低影响，我们一开始决定采用 *protector cell* 的机制。这套机制让一部分 V8 假定一部分事件尚未发生， 从而跳过某些检测。在这个例子中， protector cell 确保 V8 没有添加其他 Symbol.hasInstance 的属性。如果 `@@hasInstance` 没有在其他地方添加，并且保护器完好，就可以继续调用 OrdinaryHasInstance  来实现 `instanceof`。
 
-The assumption was that no one would use this monkey-patching ability for `instanceof` anytime soon, which buys us some time to come up with a better solution that scales well even in the presence of custom `Symbol.hasInstance` methods. But apparently this assumption was invalid, since Node.js v7 [started using `Symbol.hasInstance`](https://github.com/nodejs/node/commit/2a4b068acaa160a2d76ec5a3728e29ac6cdc715b) for their `Writable` class. This slowed down any use of `instanceof` in any Node.js by up to a [factor of 100](https://github.com/nodejs/node/issues/9634) depending on the exact usage pattern. So we had to look for mitigations of the problem, and as it turned out, there is at least an easy way to avoid depending on the global protector cell for the optimizing compilers Crankshaft and TurboFan, and so we got that fixed with [crrev.com/2504263004](https://codereview.chromium.org/2504263004) and [crrev.com/2511223003](https://codereview.chromium.org/2511223003).
+如果短期内没人使用这种补丁版的 `instanceof` , 那就相当于为实现伸缩性良好的、自定义的`Symbol.hasInstance`匀出了时间。然而这是不可能的，我们只能继续寻找缓解问题的方法。功夫不负有心人，有一种简单的方式能够优化 Crankshaft 和 TurboFan，并且不依赖与全局的 protector cell，我们依此顺利解决了这个问题，issue记录如在 [crrev.com/2504263004](https://codereview.chromium.org/2504263004) 和[crrev.com/2511223003](https://codereview.chromium.org/2511223003) 中。
```

Comment:
since Node.js v7 [started using `Symbol.hasInstance`]for their `Writable` class. This slowed down any use of `instanceof` in any Node.js by up to a [factor of 100]depending on the exact usage pattern. 
这句话是不是都没有翻译出来